contrib/babelfishpg_common/src/datetime2.c

```diff
@@ -55,6 +255,19 @@ datetime2_in_str(char *str, int32 typmod)
 	char	   *field[MAXDATEFIELDS];
 	int			ftype[MAXDATEFIELDS];
 	char		workbuf[MAXDATELEN + MAXDATEFIELDS];
+	bool		contains_extra_spaces = false;
+	bool		is_year_set = false;
+	char		*modified_str = str;
+	bool		dump_restore = false;
+	const char *babelfish_dump_restore = GetConfigOption("babelfishpg_tsql.dump_restore", true, false);
+
+	if (babelfish_dump_restore &&
+		 strncmp(babelfish_dump_restore, "on", 2) == 0)
+		 dump_restore = true;
```

Comment:
Why don't we check for this in the respective helper function instead of passing it as an argument?